server/src/com/gip/xyna/xnwh/persistence/xmom/UpdateGenerator.java

```diff
@@ -126,53 +117,172 @@ public boolean equals(Object obj) {
     
     @Override
     public int hashCode() {
-      return updateStatement.getQualifiedColumn().hashCode();
+      int hash = 0;
+      for (QualifiedStorableColumnInformation column : updateStatement.getQualifiedColumn()) {
+        hash += column.hashCode();
+      }
+      return hash;
+    }
+
+
+    @Override
+    public int compareTo(UpdateGeneration o) {
+      return compare(this, o);
     }
     
+    private static int compare(UpdateGeneration u1, UpdateGeneration u2) {
+      if(u1.listIdxs.size() != u2.listIdxs.size()) {
+        return u1.listIdxs.size() - u2.listIdxs.size() ;
+      }
+      
+      for(int i=0; i< u1.listIdxs.size(); i++) {
+        if(u1.listIdxs.get(i) != u2.listIdxs.get(i)) {
```

Comment:
you should probably use equals here instead, because the listIdx are Integers, not ints.